File: src/main/java/ar/edu/unq/spring/modelo/exception/MuchoPesoException.java
package ar.edu.unq.spring.modelo.exception;

import ar.edu.unq.spring.modelo.Personaje;
import ar.edu.unq.spring.modelo.Item;

public class MuchoPesoException extends RuntimeException {

    private static final long serialVersionUID = 1L;

    private final Personaje personaje;
    private final Item item;

    public MuchoPesoException(Personaje personaje, Item item) {
        this.personaje = personaje;
        this.item = item;
    }

    @Override
    public String getMessage() {
        return "El personaje [" + personaje + "] no puede recoger [" + item + "] porque carga mucho peso ya";
    }
}

File: src/main/kotlin/ar/edu/unq/spring/modelo/exception/MuchoPesoException.kt
package ar.edu.unq.spring.modelo.exception

import ar.edu.unq.spring.modelo.Item
import ar.edu.unq.spring.modelo.Personaje

class MuchoPesoException(private val personaje: Personaje, private val item: Item) : RuntimeException() {

    override val message: String?
        get() = "El personaje [$personaje] no puede recoger [$item] porque cagar mucho peso ya"

    companion object {

        private val serialVersionUID = 1L
    }

}
